src/commands/plugins.rs

```diff
@@ -117,6 +123,132 @@ impl Uninstall {
     }
 }
 
+#[derive(Parser, Debug)]
+pub struct Upgrade {
+    /// Name of Spin plugin to upgrade.
+    #[clap(
+        name = "PLUGIN_NAME",
+        conflicts_with = "ALL",
+        required_unless_present_any = ["ALL"],
+    )]
+    pub name: Option<String>,
+    /// Upgrade all plugins
+    #[clap(
+        short = 'a',
+        long = "all",
+        name = "ALL",
+        conflicts_with = "PLUGIN_NAME",
+        takes_value = false
+    )]
+    pub all: bool,
+    /// Source of local manifest file
+    #[clap(
+        name = "LOCAL_PLUGIN_MANIFEST",
+        short = 'f',
+        long = "file",
+        conflicts_with = "REMOTE_PLUGIN_MANIFEST",
+        conflicts_with = "PLUGIN_NAME"
+    )]
+    pub local_manifest_src: Option<PathBuf>,
+    /// Source of remote manifest file
+    #[clap(
+        name = "REMOTE_PLUGIN_MANIFEST",
+        short = 'u',
+        long = "url",
+        conflicts_with = "LOCAL_PLUGIN_MANIFEST",
+        conflicts_with = "PLUGIN_NAME"
+    )]
+    pub remote_manifest_src: Option<Url>,
+    /// skips prompt to accept the installation of the plugin.
+    #[clap(short = 'y', long = "yes", takes_value = false)]
+    pub yes_to_all: bool,
+    #[clap(
+        long = "version",
+        short = 'v',
+        conflicts_with = "REMOTE_PLUGIN_MANIFEST",
+        conflicts_with = "LOCAL_PLUGIN_MANIFEST",
+        requires("PLUGIN_NAME")
+    )]
+    /// Specify a particular version of the plugin to be installed from the centralized plugin repository
+    pub version: Option<Version>,
+    /// Allow downgrading a plugin's version
+    #[clap(short = 'd', long = "downgrade", takes_value = false)]
+    pub downgrade: bool,
+}
+
+impl Upgrade {
+    pub async fn run(self) -> Result<()> {
+        let plugins_dir = get_spin_plugins_directory()?;
+        let spin_version = env!("VERGEN_BUILD_SEMVER");
+
+        if self.all {
+            // Install the latest of all currently installed plugins
+            for plugin in std::fs::read_dir(&plugins_dir.join(PLUGIN_MANIFESTS_DIRECTORY_NAME))? {
```

Comment:
There is a potential for the command to be run before any plugins are installed. The `manifests` directory is only created after the first time someone installs a plugin. Therefore, in those cases, this would return `Error: No such file or directory (os error 2)` as the error.

Reply:
Good point! added a directory check before all other actions on upgrades: 2ad8e0b134c66